open-craft/opencraft
# BB-1876: Add frontend automatic build and deployment WIP

This PR set's up automatic deployment of the new Ocim static frontend. It does this:
1. Runs tests.
2. Builds static files.
3. Waits for user approval to run deployment.
4. Deploys to S3 and invalidates CloudFront.

**Testing instructions:**
1. Rerun the CI checks
2. Wait until the `allow-deployment` stage then click on the stage and **Approve** the deployment:
![click](https://user-images.githubusercontent.com/27893385/69183059-f8c04380-0af0-11ea-8faa-17dc43e62f71.png)
![approve](https://user-images.githubusercontent.com/27893385/69183057-f8c04380-0af0-11ea-8666-2820ffccc19d.png)
3. Watch the redeployment, check that the files were updated and a CloudFront invalidation was triggered. The AWS account is: [Ocim Stage](https://vault.opencraft.com:8200/ui/vault/secrets/secret/show/core/OpenCraft%20Tools%20:%20Resources%20:%20AWS%20:%20AWS%20S3%20Stage).
4. Check the live app: https://app.stage.console.opencraft.com/

**Reviewers:**
